core/src/pretty.rs

```diff
@@ -493,281 +592,266 @@ where
             Bool(v) => allocator.as_string(v),
             Num(n) => allocator.as_string(format!("{}", n.to_sci())),
             Str(v) => allocator.escaped_string(v).double_quotes(),
-            StrChunks(chunks) => allocator.chunks(
-                chunks,
-                if chunks.len() > 1 {
-                    StringRenderStyle::Multiline
-                } else {
-                    StringRenderStyle::Monoline
-                },
-            ),
+            StrChunks(chunks) => allocator.chunks(chunks, StringRenderStyle::Multiline),
             Fun(id, rt) => {
                 let mut params = vec![id];
                 let mut rt = rt;
                 while let Fun(id, t) = rt.as_ref() {
                     params.push(id);
                     rt = t
                 }
-                allocator
-                    .text("fun")
-                    .append(allocator.space())
-                    .append(allocator.intersperse(
+                docs![
+                    allocator,
+                    "fun",
+                    allocator.line(),
+                    allocator.intersperse(
                         params.iter().map(|p| allocator.as_string(p)),
-                        allocator.space(),
-                    ))
-                    .append(allocator.space())
-                    .append(allocator.text("=>"))
-                    .append(allocator.softline())
-                    .append(rt.to_owned().pretty(allocator).nest(2))
+                        allocator.line()
+                    ),
+                    allocator.line(),
+                    "=>",
+                    allocator.line(),
+                    rt
+                ]
+                .nest(2)
+                .group()
             }
             // TODO Pattern destructuring to implement.
```

Comment:
Is this TODO still to be done?

Reply:
It should be done, already. I'll get rid of the comment.